src/components/PersonPage/PersonContainer.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from "react";
+import Person from "../../models/Person";
+import { PersonService } from "../../networking/PersonService";
+import "@mozilla-protocol/core/protocol/css/protocol.css";
+
+type PersonContainerProps = {
+  /** optional person overrides fetch for testing */
+  testPerson?: Person;
+  testPersonId?: string;
+};
+
+const PersonContainer = ({ testPerson, testPersonId }: PersonContainerProps) => {
+  // 283ffb6c5a48
+  const [person, setPerson] = useState<Person | undefined>(testPerson);
+  const [error, setError] = useState<Error | undefined>(undefined);
+
+  useEffect(() => {
+    let personService = new PersonService();
+    const fetchData = async () => {
+      const slug = window.location.href.substring(window.location.href.lastIndexOf("/") + 1);
```

Comment:
We had a request a long time ago to have both the id and router string be available. i.e. both some sha value and `/lorena-gonzalez` for example. That value _should_ always be available for a person and it _should_ always be unique. Is there a way to check for the "router_string" in addition to the id?

Also, I know this PR is just focused on the networking layer but just curious.

Reply:
That sounds like it would be very difficult to distinguish a `router_string` vs the ID of the person object, if they are both strings and live in the same place (the slug).

Reply:
Check a substring of the slug?

`id--{some_id}` vs `route--{some_router_string}` maybe?

Reply:
Wouldn't that require that all IDs and all `router_strings` have prefixes appended to them?

Reply:
Why not in this slug parsing code check for those prefixes?

If it's using the `id--` prefix then do exactly what you are doing now but strip the prefix first.

If it's using the `route--` prefix then query for docs with that `router_string`, if more than one or none, error. if just one then use the data received?

Reply:
Since we are using react-router-dom, we should eventually use react-router-dom `useLocation` to get the slug from its [location](https://reactrouter.com/web/api/location). But this isn't particularly important for this PR.

Regarding prefix, we could switch to `/person?id=<person-id>` or `/person?router_string=<router-string>`, instead of `/person/<slug>`

Reply:
You could parse the slug to determine the prefix too.

Reply:
I switched to useLocation for now.  We'll refactor as necessary.